go/libraries/doltcore/sqle/database.go

```diff
@@ -142,7 +142,17 @@ func (db *Database) GetTableInsensitive(ctx context.Context, tblName string) (sq
 }
 
 func (db *Database) GetTableNames(ctx context.Context) ([]string, error) {
-	return db.root.GetTableNames(ctx)
+	tblNames, err := db.root.GetTableNames(ctx)
+	if err != nil {
+		return nil, err
+	}
+	for i, tbl := range tblNames {
+		if tbl == doltdb.DocTableName {
+			tblNames = append(tblNames[:i], tblNames[i+1:]...)
```

Comment:
Instead of setting and breaking here, just return the result of append.

Also does this work when dolt_docs is the last element of the list?

You might want to factor out a method called filterDoltInternalTables or similar, that takes the table names and returns a copy with the _dolt* tables filtered out.

Reply:
Resolved: https://github.com/liquidata-inc/dolt/pull/280/commits/fc2e09aad5f6b2267c341262635a0aa48b76b2cf#diff-ff4ec6598b401e530eb84ad72ea62a9fR154